proxy/auth.go

```diff
@@ -19,19 +19,17 @@ type BasicAuth struct {
 	password string
 }
 
-func (auth *BasicAuth) Set(req *http.Request) error {
+func (auth *BasicAuth) Set(req *http.Request) {
```

Comment:
I can't remember why we had `error` here, but there may have been a good reason. @viveksyngh can you speak to this?

Reply:
As I recall. Idea was to allow an interface with which we can use different authentication providers e.g the token can be obtained from a file or something or an API. In such case we might we might encounter some errors which can be returned by the method.

Reply:
If we are not checking for errors in where we are setting the auth then it might be bug

Reply:
for basic auth (req.SetBasicAuth()) AND barerAuth (header.set) the funcs dont return an error, so there's nothing to check

Reply:
Happy to put `error` back on, just spotted we didnt use it anywhere